lib/python/ray/services.py

```diff
@@ -62,33 +91,34 @@ def cleanup():
   processes that were started by this services module. Driver processes are
   started and disconnected by worker.py.
   """
-  global all_processes
   successfully_shut_down = True
   # Terminate the processes in reverse order.
-  for p in all_processes[::-1]:
-    if p.poll() is not None: # process has already terminated
-      continue
-    if RUN_PHOTON_PROFILER or RUN_PLASMA_MANAGER_PROFILER or RUN_PLASMA_STORE_PROFILER:
-      os.kill(p.pid, signal.SIGINT) # Give process signal to write profiler data.
-      time.sleep(0.1) # Wait for profiling data to be written.
-    p.kill()
-    time.sleep(0.05) # is this necessary?
-    if p.poll() is not None:
-      continue
-    p.terminate()
-    time.sleep(0.05) # is this necessary?
-    if p.poll is not None:
-      continue
-    successfully_shut_down = False
+  process_types = ["workers", "local_schedulers", "plasma_managers",
```

Comment:
remove this and the assertion and replace the thing iterated over in the loop with all_processes.keys()